mypaint-brush.c

```diff
@@ -646,6 +646,29 @@ smallest_angular_difference(float a, float b)
 
     float base_radius = expf(mypaint_mapping_get_base_value(self->settings[MYPAINT_BRUSH_SETTING_RADIUS_LOGARITHMIC]));
 
+	if (self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_X]) {
+		x += self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_X] * base_radius;	
+	}
+	
+	if (self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_Y]) {
+		y += self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_Y] * base_radius;	
+	}
+	
+	if (self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_ANGLE]) {
+		x += base_radius * self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_ANGLE] * self->states[MYPAINT_BRUSH_STATE_DIRECTION_DY];
+		y += base_radius * self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_ANGLE] * -self->states[MYPAINT_BRUSH_STATE_DIRECTION_DX];	
+	}
+	
+	if (self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_ANGLE_2]) {
+		if (self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_ANGLE_2] < 0) {
+			self->settings_value[MYPAINT_BRUSH_SETTING_OFFSET_ANGLE_2] = 0;	
+		}
+		static int sign = +1;
```

Comment:
If there is more than one brush, wouldn't a lot of other stuff break, too?  Like speed, direction, etc?  Lots of things seems to hold a state that is only relevant for a particular brush, right?  Or maybe I'm missing something.  Well, I forked rbreu's branch and I'm going to add the flip state that @martinxyz suggested, clean it up a bit and make a pull request towards Anti-Art (I think that's what I should do, instead of making a whole new pull against master)